tensorrt_llm/_torch/auto_deploy/transformations/transform.py

```diff
@@ -47,10 +47,11 @@ def __init__(
         self.factory = factory
         self.attn_backend = ad_config.attn_backend
         self.mla_backend = ad_config.mla_backend
-        # TODO (lliebenwein): let's split up the compile backend to separately handle cuda graph
-        # and torch compile so we can follow the PyTorchConfig here and enable it separately.
+
         self.ad_config = ad_config
-        if ad_config.use_cuda_graph or ad_config.torch_compile_enabled:
+        if ad_config.torch_compile_enabled:
+            compile_backend = "torch-compile"
+        elif ad_config.use_cuda_graph:
             compile_backend = "torch-opt"
         else:
             compile_backend = "torch-simple"
```

Comment:
How about we make it fully compatible with the pytorch config at this point? 

We can have a fourth backend `torch-cudagraph` that only applies cudagraph and doesn't apply torch.compile:

```suggestion
        if ad_config.torch_compile_enabled and ad_config.use_cudagraph:
            compile_backend = "torch-opt"
        elif ad_config.use_cuda_graph:
            compile_backend = "torch-cudagraph"
         elif ad_config.torch_compile_enabled:
            compile_backend = "torch-compile"
        else:
            compile_backend = "torch-simple"
```

Reply:
In the manual workflow, it is possible to apply cudagraphs without compile, but that doesn't necessarily make a whole lot of sense for AutoDeploy, I think. That is, I am not sure I see any practical need for a compilation flow where compile (the easy) is skipped but cudagraph (the tough) is applied. Do you?

Reply:
the perf regressions in the manual workflow when invoking compile is because of the additional graph breaks. Those, by definition, won't happen in AutoDeploy since we front-load graph capture. But I do see the advantages for debugging. I'll add another compile backend: torch-cudagraph